File: LeetCode_Algorithm/258(디지털 루트)/258.js
var addDigits = function(num) {
    
  // sol1 82~93ms 76~63%
  var num = String(num)
  while (num.length != 1) {
    let ans = 0
    for (let n of num){
      ans += parseInt(n)
    }
    num = String(ans)
  }
  return num

  // digital root
};

console.log(addDigits(38))

File: LeetCode_Algorithm/258(디지털 루트)/258.py
class Solution:
    def addDigits(self, num: int) -> int:

        # sol1 55~70ms 32~7%
        num = str(num)
        while len(num) != 1:
            ans = 0
            for i in num:
                ans += int(i)
            num = str(ans)
        return num

        # sol2 
        # digital root
        


print(Solution().addDigits(38))